src/components/VmDetails/cards/DetailsCard/index.js

```diff
@@ -172,7 +172,9 @@ class DetailsCard extends React.Component {
     const vmClusterId = this.props.vm.getIn(['cluster', 'id'])
 
     if (prevProps.clusters !== this.props.clusters) {
-      this.setState({ clusterList: createClusterList(this.props.clusters) }) // eslint-disable-line react/no-did-update-set-state
+      const { clusters } = this.props
+      const vmDataCenterId = clusters.getIn([vmClusterId, 'dataCenterId'])
+      this.setState({ clusterList: createClusterList(clusters, vmDataCenterId) }) // eslint-disable-line react/no-did-update-set-state
```

Comment:
Just move the `vmDataCenterId` accessing code from the storageDomains check block (line 181) up to under line 172 and have `vmClusterId` and `vmDataCenterId`. That way the clusters and storageDomains can reuse the same lookup